Samples/WinMLSamplesGallery/WinMLSamplesGalleryNative/D3D12Quad.h

```diff
@@ -0,0 +1,108 @@
+#pragma once
+
+#include <stdexcept>
+#include <wincodec.h>
+
+using namespace DirectX;
+using Microsoft::WRL::ComPtr;
+
+class D3D12Quad
+{
+public:
+    D3D12Quad(UINT width, UINT height, std::wstring name);
+
+    void OnInit();
+    void OnUpdate();
+    void OnRender();
+    void OnDestroy();
+
+    // Accessors.
+    UINT GetWidth() const { return m_width; }
+    UINT GetHeight() const { return m_height; }
+    const WCHAR* GetTitle() const { return m_title.c_str(); }
+
+    ComPtr<ID3D12Resource> GetCurrentBuffer();
+    void ShowNextImage();
+
+    bool is_initialized = false;
+
+
+private:
+    static const UINT FrameCount = 2;
+
+    struct Vertex
+    {
+        XMFLOAT3 position;
+        XMFLOAT2 uv;
+    };
+
+    // Pipeline objects.
+    CD3DX12_VIEWPORT m_viewport;
+    CD3DX12_RECT m_scissorRect;
+    ComPtr<IDXGISwapChain3> m_swapChain;
+    ComPtr<ID3D12Device> m_device;
+    ComPtr<ID3D12Resource> m_renderTargets[FrameCount];
+    ComPtr<ID3D12CommandAllocator> m_commandAllocator;
+    ComPtr<ID3D12CommandQueue> m_commandQueue;
+    ComPtr<ID3D12RootSignature> m_rootSignature;
+    ComPtr<ID3D12DescriptorHeap> m_rtvHeap;
+    ComPtr<ID3D12DescriptorHeap> m_srvHeap;
+    ComPtr<ID3D12PipelineState> m_pipelineState;
+    ComPtr<ID3D12GraphicsCommandList> m_commandList;
+    UINT m_rtvDescriptorSize;
```

Comment:
>UINT m_rtvDescriptorSize [](http://example.com/codeflow?start=4&length=24)

We should just initialize this field here (NSDMI), rather than in the constructor `D3D12Quad::D3D12Quad`, which is more brittle to change.

`UINT m_rtvDescriptorSize = 0;`

Reply:
Done.